File: src/org/mystic/codeforces/gym100229/F.scala
package org.mystic.codeforces.gym100229

import java.io._
import java.util.StringTokenizer

object F {
  var out: PrintWriter = null
  var br: BufferedReader = null
  var st: StringTokenizer = null

  def next: String = {
    while (st == null || !st.hasMoreTokens) {
      st = new StringTokenizer(br.readLine)
    }
    return st.nextToken
  }

  def nextInt: Int = {
    return Integer.parseInt(next)
  }

  def solve = {
    val string = next
    val n = string.length
    var result = 0
    for (i <- 1 to n) {
      val prefMax = prefixFunc(string.substring(0, i).reverse).toList.max
      result += string.substring(0, i).length - prefMax
    }
    out.println(result)
  }

  def prefixFunc(string: String): Array[Int] = {
    val π = new Array[Int](string.length)
    π(0) = 0
    for (i <- 1 to string.length - 1) {
      var j = π(i - 1)
      while (j != 0 && string(j) != string(i)) {
        j = π(j - 1)
      }
      if (string(j) == string(i)) {
        π(i) = j + 1
      } else {
        π(i) = 0
      }
    }
    π
  }

  def main(args: Array[String]): Unit = {
    br = new BufferedReader(new InputStreamReader(new FileInputStream("bacon.in")))
    out = new PrintWriter(new BufferedOutputStream(new FileOutputStream("bacon.out")))
    solve
    out.close
  }
}

File: src/org/mystic/opencup/round1/F.java
package org.mystic.opencup.round1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;

public class F {

    int[] tt = null;
    int tx = 0;
    private BufferedReader reader;
    private PrintWriter writer;

    public F() throws IOException {

//        reader = new BufferedReader(new FileReader("intset.in"));
//        writer = new PrintWriter(new FileWriter("intset.out"));

        reader = new BufferedReader(new InputStreamReader(System.in));
        writer = new PrintWriter(System.out);
    }

    public static void main(String[] args) throws IOException {
        new F().solve();
    }

    int[] readInts() throws IOException {
        String[] s = reader.readLine().split(" ");
        int[] ints = new int[s.length];
        for (int i = 0; i < ints.length; i++) {
            ints[i] = Integer.parseInt(s[i]);
        }
        return ints;
    }

    int readInt() throws IOException {
        if (tt == null || tx == tt.length) {
            tt = readInts();
            tx = 0;
        }
        return tt[tx++];
    }

    private void solve() throws IOException {
        int n = readInt();
        int l = readInt();
        int m = readInt();
        writer.println((1L * l * m) / n);
        writer.flush();
    }
}
